repo2docker/buildpacks/python/__init__.py

```diff
@@ -1,139 +1,70 @@
-        the `NB_PYTHON_PREFIX` to the location of the jupyter binary.
-
-        """
-        return super().get_env() + [
-            ("VENV_PATH", "${APP_BASE}/venv"),
-            # Prefix to use for installing kernels and finding jupyter binary
-            ("NB_PYTHON_PREFIX", "${VENV_PATH}"),
-        ]
-
-    def get_path(self):
-        """Return paths (including virtual environment path) to be added to
-        the PATH environment variable.
-
-        """
-        return super().get_path() + [
-            "${VENV_PATH}/bin"
-        ]
-
-    def get_build_script_files(self):
-        """
-        Dict of files to be copied to the container image for use in building.
-
-        This is copied before the `build_scripts` & `assemble_scripts` are
-        run, so can be executed from either of them.
-
-        It's a dictionary where the key is the source file path in the host
-        system, and the value is the destination file path inside the
-        container image.
-
-        This currently adds a frozen set of Python 3 requirements to the dict
-        of files.
-
-        """
-        files = {
-            'python/requirements.frozen.txt': '/tmp/requirements.frozen.txt',
-        }
-        files.update(super().get_build_script_files())
-        return files
-
-    def get_build_scripts(self):
-        """
-        Return series of build-steps common to all Python 3 repositories.
-
-        All scripts here should be independent of contents of the repository.
-
-        This sets up:
-
-        - a directory for the virtual environment and its ownership by the
-          notebook user
-        - a Python 3 interpreter for the virtual environement
-        - a Python 3 jupyter kernel including a base set of requirements
-        - support for Jupyter widgets
-        - support for JupyterLab
-        - support for nteract
-
-        """
-        return super().get_build_scripts() + [
-            (
-                "root",
-                r"""
-                mkdir -p ${VENV_PATH} && \
-                chown -R ${NB_USER}:${NB_USER} ${VENV_PATH}
-                """
-            ),
-            (
-                "${NB_USER}",
-                r"""
-                python3 -m venv ${VENV_PATH}
-                """
-            ),
-            (
-                "${NB_USER}",
-                r"""
-                pip install --no-cache-dir -r /tmp/requirements.frozen.txt && \
-                jupyter nbextension enable --py widgetsnbextension --sys-prefix && \
-                jupyter serverextension enable --py jupyterlab --sys-prefix && \
-                jupyter serverextension enable nteract_on_jupyter --sys-prefix
-                """
-            )
-        ]
+        try:
+            with open(self.binder_path('runtime.txt')) as f:
+                runtime = f.read().strip()
+        except FileNotFoundError:
+            runtime = ''
+
+        if not runtime.startswith('python-'):
```

Comment:
do we have any future plan to support multiple languages with `runtime.txt`? (e.g. both python and R).